src/napari_spatialdata/_view.py

```diff
@@ -256,12 +270,57 @@ def _select_layer(self) -> None:
         else:
             return
 
+    def _update_adata(self):
+        if (table_name := self.table_name_widget.currentText()) == "":
+            return
+        layer = self._viewer.layers.selection.active
+        if sdata := layer.metadata.get("sdata"):
+            element_name = layer.metadata.get("name")
+            table = sdata[table_name]
+            adata = table[table.obs[table.uns["spatialdata_attrs"]["region_key"]] == element_name]
+            layer.metadata["adata"] = adata
+
+        if layer is not None and "adata" in layer.metadata:
+            with self.model.events.adata.blocker():
+                self.model.adata = layer.metadata["adata"]
+
+        if self.model.adata.shape == (0, 0):
+            return
+
+        if "spatial" in self.model.adata.obsm:
+            self.model.coordinates = np.insert(
+                self.model.adata.obsm[Key.obsm.spatial][:, ::-1][:, :2], 0, values=0, axis=1
+            )
+
+        self.model.spot_diameter = np.array([0.0, 10.0, 10.0])
+        self.model._labels_key = layer.metadata["region_key"] if isinstance(layer, Labels) else None
```

Comment:
Ok for now, but we should probably get rid of the legacy `label_key` in a follow-up PR.

Reply:
Tracked here: https://github.com/scverse/napari-spatialdata/issues/180